lisa/nic.py

```diff
@@ -456,3 +515,57 @@ def _get_default_nic(self) -> None:
         ).is_true()
         self.default_nic: str = default_interface_name
         self.default_nic_route = dev_match.group()
+
+    def get_used_module(self) -> str:
+        lspci = self._node.tools[Lspci]
+        pci_devices = lspci.get_devices_by_type(
+            constants.DEVICE_TYPE_SRIOV, force_run=True
+        )
+        # there will not be multiple Mellanox types in one VM
+        # get the used module using any one of sriov device
+        used_module = lspci.get_used_module(pci_devices[0].slot)
+
+        if not used_module:
```

Comment:
The all logic should be proceeded in some initialization or load method. other places just check the loaded class info. It can make the logic clearer, and easy to maintain.